src/devtools/useAtomsSnapshot.ts

```diff
@@ -0,0 +1,38 @@
+import { useCallback, useContext, useMemo } from 'react'
+import { SECRET_INTERNAL_getStoreContext as getStoreContext } from 'jotai'
+import { AtomState, State, subscribeAtom } from '../core/vanilla'
+import { RegisteredAtomsContext } from '../core/contexts'
+import { useMutableSource } from '../core/useMutableSource'
+import { AnyAtom } from '../core/types'
+
+type AtomsSnapshot = Map<AnyAtom, unknown>
+
+export function useAtomsSnapshot(): AtomsSnapshot {
+  const StoreContext = getStoreContext()
+  const [mutableSource] = useContext(StoreContext)
+  const atoms = useContext(RegisteredAtomsContext)
+
+  const subscribe = useCallback(
+    (state: State, callback: () => void) => {
+      // FIXME we don't need to resubscribe, just need to subscribe for new one
+      const unsubs = atoms.map((atom) => subscribeAtom(state, atom, callback))
+      return () => {
+        unsubs.forEach((unsub) => unsub())
+      }
+    },
+    [atoms]
+  )
+  const state: State = useMutableSource(mutableSource, getState, subscribe)
+
+  return useMemo(() => {
+    const atomToAtomStateTuples = atoms
```

Comment:
```suggestion
    const atomToAtomValueTuples = atoms
```

I'd like to call it "atom value", because `AtomState` means something else.